service/policy/kasregistry/key_access_server_registry.proto

```diff
@@ -114,39 +116,107 @@ message KeyAccessServerGrants {
   repeated GrantedPolicyObject value_grants = 4;
 }
 
-// TODO: optional validation below should be through a custom validator, which is too bleeding edge at present without full plugin support
+/*
+  KEY MANAGEMENT
+*/
+
+message CreateKeyRequest {
+  // Required
+  string kas_id = 1 [ (buf.validate.field).string.uuid = true ];
+
+  // Required
+  KasPublicKey key = 2;
+
+   // Common metadata
+  common.MetadataMutable metadata = 100;
+}
+
+message CreateKeyResponse { Key key = 1; }
+
+message GetKeyRequest { string id = 1; }
+
+message GetKeyResponse { Key key = 1; }
+
+message ListKeysRequest {
+  // Optional
+  string kas_id = 1;
+  // Optional
+  string kas_name = 2;
+  // Optional
+  string kas_uri = 3;
+  // Optional
+  policy.PageRequest pagination = 10;
+}
+
+message ListKeysResponse { 
+  repeated Key keys = 1; 
 
-// LIST of KAS Grants returns flat response of grants to all policy objects. It does not
-// employ selectors for grants to specific policy objects or build the attribute tree relation.
-// If grants to a known namespace, attribute, or value are needed, use the respective
-// GET request to the specific policy object.
+  policy.PageResponse pagination = 10;
+}
+
+message UpdateKeyRequest {
+  // Required
+  string id = 1;
+  bool active = 2;
```

Comment:
Is the supported update functionality strictly deactivation?

Reply:
Should we make the paths clearer between deactivation, reactivation, and a true update (change of KID, alg, or PEM) even if those are all unsafe?

Reply:
I think adding deactivation and reactivation makes sense. 

@cassandrabailey293 @jp-ayyappan When managing public keys should we allow modification of that resource or should we force an admin to create a new key so they can retain it for audit purposes in the database.